app_test.go

```diff
@@ -100,6 +100,58 @@ func TestInvokes(t *testing.T) {
 	})
 }
 
+func TestError(t *testing.T) {
+	t.Run("NilErrorOption", func(t *testing.T) {
+		var invoked bool
+
+		app := fxtest.New(t,
+			Error(nil),
+			Invoke(func() { invoked = true }), // should not be called
+		)
+		err := app.Err()
+		require.NoError(t, err)
+		assert.True(t, invoked)
+	})
+
+	t.Run("SingleErrorOption", func(t *testing.T) {
+		var invoked bool
+
+		app := fxtest.New(t,
+			Error(errors.New("module failure")),
+			Invoke(func() { invoked = true }), // should not be called
+		)
+		err := app.Err()
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "module failure")
+		assert.False(t, invoked)
+	})
+
+	t.Run("MultipleErrorOption", func(t *testing.T) {
+		type A struct{}
+		var provided, invoked bool
+
+		app := fxtest.New(t,
+			Provide(
+				func() A { // should not be called
+					provided = true
+					return A{}
+				},
+			),
```

Comment:
Whitespace is a little odd here - can we put the `func() A` on the same line as the `Provide(`?